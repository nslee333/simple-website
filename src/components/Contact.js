export default function Contact() {
  const renderForm = () => {
    return (
      <>
        <form name="contact" data-netlify="true" method="post">
          <input type="hidden" name="form-name" value="contact" />
          <label>
            <div hidden>name</div>
            <div>
              <input
                className="my-2 
                          block 
                          w-full 
                          rounded-md 
                          p-2"
                type="text"
                name="name"
                placeholder="Full Name"
                required={true}
              />
            </div>
          </label>
          <label>
            <div hidden>email</div>
            <div className="my-3">
              <input
                className="my-2 
                          block 
                          w-full 
                          rounded-md 
                          p-2"
                type="email"
                name="email"
                placeholder="Email"
                required={true}
              />
            </div>
          </label>
          <label>
            <div
              className="my-4
                        block
                        w-full"
            >
              <textarea
                className="my-2 
                            h-[10rem] 
                            w-full 
                            resize-none 
                            rounded-md 
                            p-2"
                type="text"
                name="message"
                placeholder="..."
                required={true}
              />
            </div>
          </label>
          <button
            className="shadow-grey 
                        mb-2 
                        ml-2 
                        mt-4 
                        block
                        h-[2rem] 
                        w-[7rem] 
                        rounded-md 
                        bg-white 
                        text-xl 
                        shadow
                        hover:translate-y-0.5
                        hover:origin-bottom 
                        hover:shadow-none"
            type="submit"
          >
            Send
          </button>
        </form>
      </>
    );
  };

  return (
    <>
      <div className="flex justify-center p-2.5">
        <div
          className="mx-2 
                      mt-5 
                      w-[32rem] 
                      min-w-[18.85rem] 
                      rounded-md
                      bg-white "
          id="contact"
        >
          <h1
            className="m-5 
                        flex 
                        justify-center 
                        text-2xl"
          >
            Contact Us...
          </h1>
          <div
            className="m-5 
                        rounded-md
                        bg-neutral-300 
                        p-5"
          >
            {renderForm()}
          </div>
        </div>
      </div>
    </>
  );
}
