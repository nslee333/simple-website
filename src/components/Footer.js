import React from "react";
import github_logo from "../../public/images/github_logo_white.svg";
import Image from "next/image";

export default function Footer() {
  return (
    <>
      <div className="m-5 flex justify-center p-10">
        <a href="https://github.com/BendJS/simple-website" target="_blank">
          <div className="text-white">
            <div>Contribute to this website</div>
            <div className="flex justify-center p-2">
              <Image src={github_logo} alt="GitHub Logo" className="w-[5rem]" />
            </div>
          </div>
        </a>
      </div>
    </>
  );
}
