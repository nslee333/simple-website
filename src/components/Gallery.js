import React from "react";
import { Navigation, Pagination } from "swiper";
import { Swiper, SwiperSlide } from "swiper/react";
import Image from "next/image";
import { urlFor } from "../pages";

import "swiper/swiper-bundle.min.css";
import "swiper/swiper.min.css";

export default function Gallery(images) {
  const imagesArr = images.props;
  return (
    <div className="my-10 flex justify-center ">
      <div
        className="border-white-200
                    rounded-md
                    border-8
                    xs:w-[19rem]
                    sm:w-[30rem]
                    md:w-[40rem]
                    lg:w-[50rem]"
      >
        <Swiper
          modules={[Navigation, Pagination]}
          spaceBetween={50}
          slidesPerView={1}
          navigation={true}
          pagination={{ clickable: true }}
          style={{
            "--swiper-pagination-color": "#ffffff",
            "--swiper-pagination-bullet-inactive-color": "#999999",
            "--swiper-pagination-bullet-inactive-opacity": "1",
            "--swiper-pagination-bullet-size": "10px",
            "--swiper-pagination-bottom": "25px",
            "--swiper-navigation-color": "#0",
          }}
        >
          {imagesArr ? (
            imagesArr.map((image, index) => {
              return (
                <SwiperSlide key={index}>
                  <div className="aspect" key={index}>
                    <Image
                      src={urlFor(imagesArr[index].image).url()}
                      alt={`bendjs ${index}`}
                      key={index}
                      fill
                      priority
                    />
                  </div>
                </SwiperSlide>
              );
            })
          ) : (
            <div></div>
          )}
        </Swiper>
      </div>
    </div>
  );
}
