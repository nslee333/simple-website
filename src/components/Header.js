import Image from "next/image";
import bendjs_logo from "../../public/images/bendjs_logo_square.png";
import Head from "next/head";

export default function Header() {
  return (
    <div className="flex justify-center">
      <Head>
        <title>BendJS Website</title>
      </Head>
      <div className="mx-5 my-5 w-[50rem]">
        <a href="/">
          <Image
            className="w-[8rem] rounded-lg"
            src={bendjs_logo}
            alt="bendjs logo"
          />
        </a>
      </div>
    </div>
  );
}
