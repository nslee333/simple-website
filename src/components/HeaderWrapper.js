import styled from 'styled-components';
import hero_image from "../../public/images/hero_image.jpg";

export const StyledWrapper = styled.div`
  background-image: url(${hero_image});
  // background-position: top; 
`;
