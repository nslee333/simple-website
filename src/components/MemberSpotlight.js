import Image from "next/image";
import { createClient } from "next-sanity";
import { urlFor } from "../pages";

export default function MemberSpotlight(member) {
  return (
    <>
      {member.props ? (
        <div className="sm:py-15 bg-black p-5 xs:my-0 xs:flex-wrap md:block pb-12">
          <h5
            className="mx-1
                      flex
                      justify-center 
                      p-5 
                      text-white 
                      xs:text-2xl 
                      sm:text-3xl"
          >
            Member Spotlight
          </h5>

          <div className="xs:mx-1 xs:flex xs:justify-center ">
            <div className="align-center block justify-around xs:p-0 md:flex md:w-[50rem] lg:w-[70rem]">
              <div>
                <div
                  className=" rounded-md
                            bg-[#7FE7D7]
                              p-0.5 
                              align-middle

                              xs:my-2 
                              xs:max-h-[12.5rem]
                              xs:min-h-[9rem]
                              xs:min-w-[5rem]

                              xs:max-w-[12.5rem]
                              sm:m-2
                              sm:min-w-[12.5rem]
                              sm:max-w-[12.5rem]

                              md:m-5
                              md:max-h-[15.3rem]

                              md:min-h-[15rem]
                              md:min-w-[15rem] 
                              md:max-w-[16rem]
                              min-[964px]:min-h-[13rem]

                              min-[964px]:min-w-[14rem]
                              lg:max-h-[30rem]
                              lg:max-w-[30rem]"
                >
                  <div
                    className="m-5
                                h-[22rem]
                                max-h-[20rem]
                                min-h-[18rem]

                                w-[20rem]
                                min-w-[14rem]
                                max-w-[25rem] 
                                rounded-md

                                bg-black 
                                xs:m-2
                                xs:max-h-[11rem]
                                xs:min-h-[11rem] 

                                xs:min-w-[11rem]
                                xs:max-w-[11rem]
                                xs:p-5
                                md:max-h-[14rem]

                                md:min-h-[14rem] 
                                md:min-w-[14rem]
                                md:max-w-[14rem]
                                lg:max-h-[20rem]

                                lg:max-w-[20rem]
                                "
                  >
                    <Image
                      className="content-center rounded-md"
                      src={urlFor(member.props[0].memberImage).url()}
                      alt="member profile"
                      width={300}
                      height={175}
                    />
                  </div>
                </div>
              </div>
              <div
                className="h-auto
                            min-h-[10rem] 
                            min-w-[10rem] 
                            max-w-[30rem]
                            rounded-md
                            bg-[#7FE7D7] 
                            xs:m-0 
                            xs:p-0.5
                            sm:m-2
                            md:m-5
                            lg:max-w-[40rem]"
              >
                <div className="m-5 
                                rounded-md 
                                bg-black 
                                xs:m-2 
                                xs:p-2 

                                xs:min-h-[15rem]
                                xs:min-w-[16rem] 

                                sm:min-h-[17rem]
                                sm:min-w-[25rem] 
                                
                                md:min-h-[17rem]
                                md:min-w-[25rem]

                                lg:min-h-[20rem]
                                lg:min-w-[35rem]
                                "
                  >
                  <h5 className="sm:text-md my-2 text-white sm:text-lg md:text-xl lg:text-2xl">
                    {member.props[0].memberName}
                  </h5>
                  <div className="md:text-md md:text-md text-white xs:text-sm lg:text-lg">
                    {member.props[0].memberBio}
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      ) : (
        <div></div>
      )}
    </>
  );
}

export const client = createClient({
  projectId: process.env.NEXT_PUBLIC_SANITY_PROJECT_ID,
  dataset: process.env.NEXT_PUBLIC_SANITY_DATASET,
  apiVersion: "2023-03-05",
  useCdn: false,
});

export async function getStaticProps() {
  const member = await client.fetch(`*[_type == 'MemberSpotlight']`);

  return {
    props: {
      member,
    },
  };
}
