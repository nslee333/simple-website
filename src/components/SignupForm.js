export default function SignupForm() {
  return (
    <div className="grid  w-[100%] rounded-md bg-neutral-300 p-2">
      <form
        className="flex-wrap"
        name="newsletter"
        data-netlify="true"
        method="post"
      >
        <input
          className="m-2 block rounded-md p-2"
          type="hidden"
          name="form-name"
          value="newsletter"
        />
        <label>
          <div hidden>name</div>
          <input
            className="m-2 block w-[95%] min-w-[13.25rem] rounded-md p-2"
            type="text"
            name="name"
            placeholder="Full Name"
            required={true}
          />
        </label>
        <label>
          <div hidden>email</div>
          <input
            className="m-2 block w-[95%] min-w-[13.25rem] rounded-md p-2"
            type="email"
            name="email"
            placeholder="Email"
            required={true}
          />
        </label>
        <button
          className="shadow-grey 
                      mb-2 
                      ml-2 
                      mt-4
                      block 
                      h-[2rem] 
                      w-[20%] 
                      min-w-[7rem] 
                      rounded-md 
                      bg-white 
                      text-xl 
                      shadow 
                      hover:origin-bottom 
                      hover:translate-y-0.5
                      hover:shadow-none"
          type="submit"
        >
          Sign up
        </button>
      </form>
    </div>
  );
}
