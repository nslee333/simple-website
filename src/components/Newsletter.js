import SignupForm from "./SignupForm";

export default function Newsletter() {
  return (
    <div className="m-5 h-auto w-[42rem] min-w-[18.85rem] rounded-md bg-white p-5 sm:h-[25rem]">
      <div>
        <h3 className="p-2 text-3xl italic">Newsletter</h3>
        <h4 className="p-2 text-2xl italic">Stay up to date</h4>
        <div className="p-2 italic">
          Sign up for our monthly newsletter where we share local events in
          Bend, tech news and our community member spotlight!
        </div>
      </div>
      <div className="mx-2 my-2 flex">
        <SignupForm />
      </div>
    </div>
  );
}
