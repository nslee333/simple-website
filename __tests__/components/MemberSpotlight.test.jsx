import { render } from "@testing-library/react";
import MemberSpotlight from "../../src/components/MemberSpotlight";
import client from "../../src/components/MemberSpotlight";
import { getStaticProps } from "../../src/components/MemberSpotlight";
import renderer from "react-test-renderer";

import * as matchers from "jest-extended";
expect.extend(matchers);

test("MemberSpotlight renders without crashing", () => {
  render(<MemberSpotlight />);
});

test("Memberspotlight renders correctly according to snapshot.", () => {
  const tree = renderer.create(<MemberSpotlight />).toJSON();

  expect(tree).toMatchSnapshot();
});

test("Client is defined", () => {
  const clientInstance = client;

  expect(clientInstance).toBeDefined();
});

test("getStaticProps is defined.", () => {
  const props = getStaticProps();

  expect(props).toBeDefined();
});
