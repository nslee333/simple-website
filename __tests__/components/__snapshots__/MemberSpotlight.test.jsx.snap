// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Memberspotlight renders correctly according to snapshot. 1`] = `<div />`;
